### PR TITLE
SDIT-2050: ✨ Try to get actions to upload even if success and slack notification always

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: .github/actions/slack_codescan_notification/action.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@ inputs:
   title:
     description: "Title of the failed workflow"
     required: true
+  warningOnly:
```
Comment (petergphillips) [inline .github/actions/slack_codescan_notification/action.yml:7]: this action now handles both warning and error scenarios so don't need different configuration for them

File: .github/workflows/security_owasp.yml [hunk 2]
```diff
@@ -38,49 +38,22 @@ jobs:
       id: owasp-analyse
       run: ./gradlew ${{ inputs.subproject }}:dependencyCheckAnalyze --info
       continue-on-error: true
-
-    - uses: ministryofjustice/hmpps-github-actions/.github/actions/security_owasp_reports@v0.4 # WORKFLOW_VERSION
-      if: steps.owasp-analyse.outcome == 'failure'
+    - uses: github/codeql-action/upload-sarif@v3
```
Comment (petergphillips) [inline .github/workflows/security_owasp.yml:41]: move the separate owasp reports action here as only used by this workflow anyway

File: .github/workflows/security_owasp.yml [hunk 3]
```diff
@@ -38,49 +38,22 @@ jobs:
...
-                ]
-              }
-            ]
-          }
-      env:
+        name: gradle-owasp-dependency-check-${{ inputs.subproject == '' && github.event.repository.name || inputs.subproject }}
+        path: ${{ inputs.subproject == '' && '.' || inputs.subproject }}/build/reports/dependency-check-report.html
+    - name: Gradle OWASP slack notification
+      uses: ministryofjustice/hmpps-github-actions/.github/actions/slack_codescan_notification@pgp-SDIT-2050-improve-slack-messages # WORKFLOW_VERSION
+      if: (failure() || steps.owasp-analyse.outcome == 'failure') && inputs.channel_id != 'NO_SLACK'
```
Comment (petergphillips) [inline .github/workflows/security_owasp.yml:53]: will always run if there is a failure or if the OWASP analyse task failed (as `continue-on-error` set to `true`)

File: .github/workflows/security_owasp.yml [hunk 4]
```diff
@@ -38,49 +38,22 @@ jobs:
...
+      with:
+        title: "Gradle OWASP"
+        warningOnly: ${{ steps.owasp-upload-sarif.outcome == 'success' && steps.owasp-upload-artifact.outcome == 'success' }}
```
Comment (petergphillips) [inline .github/workflows/security_owasp.yml:56]: ensure that it is only a warning if both upload steps succeeded (they will fail if the analyse fails to create a file)

File: .github/workflows/security_trivy.yml [hunk 5]
```diff
@@ -29,7 +30,7 @@ jobs:
     steps:
     - uses: actions/checkout@v4
     - name: Trivy Image Vulnerability Scanner
-      id: trivy
+      id: trivy-analyse
```
Comment (petergphillips) [inline .github/workflows/security_trivy.yml:33]: keep name consistent with gradle

**Review by james-jdgtl — APPROVED** [review]
Ready to be tagged as 0.5!

**Pull request merged**